python/cudf/cudf/api/extensions/accessor.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+from cudf.utils.docutils import docfmt_partial
+import warnings
+
+import cudf
+from pandas.core.accessor import CachedAccessor
+
+
+_docstring_register_accessor = """
+    Extends `cudf.{klass}` with custom defined accessor
+
+    Parameters
+    ----------
+    name : str
+        The name to be registered in `{klass}` for the custom accessor
+
+    Returns
+    -------
+    decorator : callable
+        Decorator function for accessor
+
+    Notes
+    -----
+    The `{klass}` object will be passed to your custom accessor upon first
+    invocation. And will be cached for future calls.
+
+    If the data passed to your accessor is of wrong datatype, you should
+    raise an `AttributeError` in consistent with other cudf methods.
+
+
+    Examples
+    --------
+    {example}
+"""
+
+_dataframe_example = """
+    In your library code:
+
+        .. code-block:: python
+
+            import cudf as gd
+
+            @gd.api.extensions.register_dataframe_accessor("point")
+            class PointsAccessor:
+                def __init__(self, obj):
+                    self._validate(obj)
+                    self._obj = obj
+
+                @staticmethod
+                def _validate(obj):
+                    cols = obj.columns
+                    if not all([vertex in cols for vertex in ["x", "y"]]):
+                        raise AttributeError("Must have vertices 'x', 'y'.")
+
+                @property
+                def bounding_box(self):
+                    xs, ys = self._obj["x"], self._obj["y"]
+                    min_x, min_y = xs.min(), ys.min()
+                    max_x, max_y = xs.max(), ys.max()
+
+                    return (min_x, min_y, max_x, max_y)
+
+    Then in user code:
+
+        .. code-block:: ipython
+
+            In [3]: df = gd.DataFrame({'x': [1,2,3,4,5,6], 'y':[7,6,5,4,3,2]})
+            In [4]: df.point.bounding_box
+            Out[4]: (1, 2, 6, 7)
```

Comment:
Let's use a plain Python prompt as that's more universal and familiar:

```suggestion
            >>> df = gd.DataFrame({'x': [1,2,3,4,5,6], 'y':[7,6,5,4,3,2]})
            >>> df.point.bounding_box
            (1, 2, 6, 7)
```